### PR TITLE
feat: New users login Flow [TAS-94]

### PR DESCRIPTION
**Pull request by femacedogomes**

### CODE REVIEW

File: src/app/(authentication)/auth/(withHeader)/(public)/register/user/FormComponent.tsx [hunk 1]
```diff
@@ -49,7 +53,7 @@ const FormComponent = () => {
           await signIn("credentials", {
             email: values.email,
             password: values.password,
-            callbackUrl: "/",
+            callbackUrl: callbackUrl ? callbackUrl : "/",
```
Comment (renanpontez) [inline src/app/(authentication)/auth/(withHeader)/(public)/register/user/FormComponent.tsx:56]: This is the same as: `callbackUrl: callbackUrl ?? "/",`

File: src/app/(authentication)/auth/(withHeader)/(public)/register/user/FormComponent.tsx [hunk 2]
```diff
@@ -49,7 +53,7 @@ const FormComponent = () => {
...
           })
             .then((resp) => {
               console.log(resp);
```
Comment (renanpontez) [inline src/app/(authentication)/auth/(withHeader)/(public)/register/user/FormComponent.tsx:59]: +1 protein bar

File: src/components/orgs/Coupon/Container.tsx [hunk 3]
```diff
@@ -200,44 +205,50 @@ const CouponContainer: React.FC<CouponContainerProps> = ({
...
+          if (error?.response?.data?.code === 404) {
+            showToastify({
+              label: "Nenhum dado encontrado. Por favor, tente novamente.",
+              type: "error",
+            });
+          }
+        });
+    } else {
+      signIn(undefined, {
+        callbackUrl: `/supplier-dashboard/${supplierId}/?coupon=${CouponId}`,
```
Comment (renanpontez) [inline src/components/orgs/Coupon/Container.tsx:249]:
```suggestion
        callbackUrl: `/supplier-dashboard/${supplierId}/?coupon=${couponId}`,
```

Please watch out case

**Review by renanpontez — APPROVED** [review]
looks good, just small non-blocking comments.

**Review by pedrohbl03 — CHANGES_REQUESTED** [review]

File: src/app/(authentication)/auth/(withHeader)/(public)/login/FormComponent.tsx [hunk 4]
```diff
@@ -43,19 +43,26 @@ const FormComponent = () => {
...
+          }
         }
 
         if (resp && resp?.error) {
           return Promise.reject(JSON.parse(resp?.error));
         }
       })
       .catch((error) => {
-        console.log(error);
+        if (error?.code === "R01") {
```
[the remark this answers is not in the document]
Comment (femacedogomes) [reply src/app/(authentication)/auth/(withHeader)/(public)/login/FormComponent.tsx:60]: thats the name of the error sended by back-end

**Pull request merged**